File: src/minigames/speed_builders/java/work/lclpnet/ap2/game/speed_builders/SpeedBuildersMiniGame.java
package work.lclpnet.ap2.game.speed_builders;

import net.minecraft.core.RegistryAccess;
import net.minecraft.resources.Identifier;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import work.lclpnet.ap2.ApConstants;
import work.lclpnet.ap2.api.game.*;

public class SpeedBuildersMiniGame implements MiniGame {

    @Override
    public boolean canBeFinale(GameStartContext context) {
        return true;
    }

    @Override
    public boolean canBePlayed(GameStartContext context) {
        return true;
    }

    @Override
    public MiniGameInstance createInstance(MiniGameHandle gameHandle) {
        return new SpeedBuildersInstance(gameHandle);
    }

    @Override
    public Identifier getId() {
        return ApConstants.identifier("speed_builders");
    }

    @Override
    public GameType getType() {
        return GameType.FFA;
    }

    @Override
    public String getAuthor() {
        return ApConstants.PERSON_LCLP;
    }

    @Override
    public ItemStack getIcon(RegistryAccess manager) {
        return new ItemStack(Items.BRICKS);
    }
}

File: src/minigames/speed_builders/src/main/kotlin/work/lclpnet/ap2/game/speed_builders/SpeedBuildersMiniGame.kt
package work.lclpnet.ap2.game.speed_builders

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.game.GameStartContext
import work.lclpnet.ap2.api.game.GameType
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.ap2.game.MiniGameFactory
import work.lclpnet.ap2.game.util.MapLevelGameFactory

class SpeedBuildersMiniGame : MiniGame {
    override val id = ApConstants.identifier("speed_builders")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_LCLP
    override fun getIcon(manager: RegistryAccess): ItemStack = ItemStack(Items.BRICKS)
    override fun canBeFinale(context: GameStartContext) = true
    override fun canBePlayed(context: GameStartContext) = true
    override fun createFactory(): MiniGameFactory = MapLevelGameFactory(::SpeedBuildersInstance)
}
